app/models/topic_tracking_state.rb

```diff
@@ -579,7 +579,7 @@ def self.trigger_post_read_count_update(post, groups, last_read_post_number, use
   def self.secure_category_group_ids(topic)
     category = topic.category
 
-    if category.read_restricted
+    if category&.read_restricted
```

Comment:
This method is responsible for returning the group_ids for the message-bus notification. I think returning "nil" when the category is nil means the message-bus notification will be public.

How about returning only the admin group if category is nil?